Updated release and index (#280)
aau-giraf/wiki: docs/Releases/2020E/2020es3r1.md

# 2020s3r1

The goal of this release was to improve the general stability of the application. By implementing as well as adding features requested by the customer.

## Completed Issues

In the following, a list of the issues completed in the different repositories will be shown.

### weekplanner

> #### Issues

Reviewer: ```suggestion

```

[364](https://github.com/aau-giraf/weekplanner/issues/364) |Warning on memory waste/security issue for API lvl 19 |[647](https://github.com/aau-giraf/weekplanner/issues/647)  
[352](https://github.com/aau-giraf/weekplanner/issues/352) |Timer is not saved when you cancel activity and then undo the cancel |
[327](https://github.com/aau-giraf/weekplanner/issues/327) |Artifacts |
[308](https://github.com/aau-giraf/weekplanner/issues/308) |Visible red line when timer has finished. |[594](https://github.com/aau-giraf/weekplanner/issues/594)  
[304](https://github.com/aau-giraf/weekplanner/issues/304) |The 'add new weekplan'-button is still available in edit-mode, but does nothing. |[560](https://github.com/aau-giraf/weekplanner/issues/560)  
[273](https://github.com/aau-giraf/weekplanner/issues/273) |As a guardian I would like to be able to delete pictograms I've uploaded so that I can remove wrongly uploaded / outdated pictograms |[606](https://github.com/aau-giraf/weekplanner/issues/606)  
[206](https://github.com/aau-giraf/weekplanner/issues/206) |API Exceptions are thrown everywhere where an api call was made previously when the log out button is pressed |

### web-api  

> #### Issues  

Reviewer: ```suggestion
```

[112](https://github.com/aau-giraf/web-api/issues/112) |Activity timer disappears after pressing 'Fortryd' after cancelling the activity. |
[93](https://github.com/aau-giraf/web-api/issues/93) |As a developer I would like to be able to run the integrations tests independent on the sample-data already being in the database |
[91](https://github.com/aau-giraf/web-api/issues/91) |Remove redundant endpoint |[176](https://github.com/aau-giraf/web-api/issues/176)  
[49](https://github.com/aau-giraf/web-api/issues/49) |Remove hardcoded values from Controller |
[47](https://github.com/aau-giraf/web-api/issues/47) |As a developer, I would like the table `Pictograms` to not include `ImageHash`. |
[19](https://github.com/aau-giraf/web-api/issues/19) |Updating order of activities when reordering in the weekplan |[180](https://github.com/aau-giraf/web-api/issues/180)  
[5](https://github.com/aau-giraf/web-api/issues/5) |Illogical endpoints |[172](https://github.com/aau-giraf/web-api/issues/172)  

### api_client  

> #### Issues  

Reviewer: ```suggestion  
```

|Number |Title |Solved by PR |
 |:---------|:---------|:---------|
[81](https://github.com/aau-giraf/api_client/issues/81) |Updating order of activities when reordering in the weekplan |[86](https://github.com/aau-giraf/api_client/issues/86)  
[80](https://github.com/aau-giraf/api_client/issues/80) |As a developer I would like an interface for the offline features |[83](https://github.com/aau-giraf/api_client/issues/83)  
[74](https://github.com/aau-giraf/api_client/issues/74) |As a developer I would like the pictogram model to have a the Id of the creating user |[75](https://github.com/aau-giraf/api_client/issues/75)  
[56](https://github.com/aau-giraf/api_client/issues/56) |Refactor the code so we don't use the Observable class |[73](https://github.com/aau-giraf/api_client/issues/73)  
[52](https://github.com/aau-giraf/api_client/issues/52) |Fundamental bug in the offline repository |
[3](https://github.com/aau-giraf/api_client/issues/3) |Implement updateIcon in user_api.dart |

## Result:

Based on the implemented issues, the following have been accomplished:

>  - Searching no longer case sensitive

Reviewer: ```suggestion
 - Searching is no longer case sensitive
```

>  - Stop buttins on timers have become restart button.

Reviewer: ```suggestion
 - Stop buttons on timers have become restart button.
```

 - Guardians can unmark completed activities

>  - Citizens can reeenter completed activities

Reviewer: ```suggestion
 - Citizens can reenter completed activities
```

 - Guardians can log out from the "Show activity" screens

>  - Moving weekplans optimized, so it no longer reloads the entire weekplan.

Reviewer: ```suggestion
 - Moving weekplans has been optimized, so it no longer reloads the entire week plan.
```

>  - Added more faster, and better notifications when errors occour.

Reviewer: ```suggestion
 - Added new and improved existing notifications when errors occur.
```
